src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -591,9 +591,7 @@ public static PSDataCollection<PSObject> GetPSDataCollection(object inputValue)
         /// <param name="second">Object to compare first to.</param>
```

Comment:
@vexx32, your last commit had 5 failures in `PowerShell-CI-windows`
SemanticVersion api tests.Error handling.Missing parts: 'NullString': '$null'
```powershell
Expected an exception, with FullyQualifiedErrorId 'FormatException' to be thrown, but the FullyQualifiedErrorId was 'PSArgumentNullException'. from D:\a\1\s\test\powershell\engine\Basic\SemanticVersion.Tests.ps1:209 char:19
    +                 { [SemanticVersion]::Parse($version) } | Should -Thro ...
    +                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\SemanticVersion.Tests.ps1: line 209
209:                 { [SemanticVersion]::Parse($version) } | Should -Throw -ErrorId "FormatException"
```
Null Representatives.Casting Behaviour.$null should be treated as $false by Where-Object
```powershell
Expected $null or empty, but got 100.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 64
64:             100 | Where-Object { $Value } | Should -BeNullOrEmpty
```
Null Representatives.Casting Behaviour.$null should be treated as $false by Where-Object
```powershell
Expected $null or empty, but got 100.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 64
64:             100 | Where-Object { $Value } | Should -BeNullOrEmpty
```
Null Representatives.Casting Behaviour.$null should cast to $false
```powershell
Expected $false, but got $true.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 58
58:             [bool]$Value | Should -BeFalse
```
Null Representatives.Casting Behaviour.$null should cast to $false
```powershell
Expected $false, but got $true.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1: line 58
58:             [bool]$Value | Should -BeFalse
```

Reply:
🤔 I see I have some work yet to do. 😄